connection/src/thick.rs

```diff
@@ -69,6 +71,8 @@ pub enum ThickClientAttestationError {
     UriConversionError(UriConversionError),
     /// Credentials provider error: {0}
     CredentialsProvider(Box<dyn CredentialsProviderError + 'static>),
+    /// Other: {0}
+    Other(String),
```

Comment:
These `Other` variants were created to handle errors calculating time. These should never occur, but technically could if the system time changes. It feels strange to make a dedicated variant for something that should never occur, but creating an `Other` variant that only has one cause also feels weird. Open to suggestions on this.

Reply:
I'm okay with `Other` for that stuff, but would also consider something like `InconsistentTime` (and then retrying on that variant would make sense)